### ISSUE TITLE
General capability to set halos on POI icons

### ISSUE THREAD

**Issue by ZeLonewolf** [ZeLonewolf/openstreetmap-americana#324]
Currently, we are not able to set halos for SVG-originated icons.  We need halos for cases when POI icons are obscured when they overlap similarly-colored features:
![image](https://user-images.githubusercontent.com/3254090/168501843-5ff9eac2-6673-4ce0-8c56-cc3dc27fd75a.png)

The mapLibre documentation indicates that this can be done with SDF icons, which can be both recolored and halo'd.

Work is needed to figure out how to add recolorable SDFs to the style and add halos.  When properly implemented, all original icon artwork can be done in monochrome.

The workaround for this is to build a border into the icon, like we do for the capital / city icons.

**Comment by claysmalley** [comment]
This may also be relevant to highway shields with simple shapes but a variety of color schemes, like the flowerpot shields of county roads in Ontario, or the various places that use a home-plate-shaped shield.

**Comment by ZeLonewolf** [comment]
I'm not sure whether or not we could make that work for shields, since we generate them in code.  That's something we can revisit once somebody understands how SDF icons work :)

**Comment by zekefarwell** [comment]
For the the city dots and capital city stars, I included halos in the graphics, probably because I couldn't find a Maplibre propery that would do it. Recolorable icons via SDF seems like a nice feature.  Looks like they are limited to one color only though, so not an option for any multicolor icons.

**Comment by ZeLonewolf** [comment]
Good point on the coloring.  Also, if there's a legitimate use case for multicolored icons, the built-in halo in the SVG remains an option for that.
